### PR TITLE
[Merged by Bors] - chore: backports for leanprover/lean4#4814 (part 24)

### PR DESCRIPTION
**Pull request by kim-em**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/LinearAlgebra/Orientation.lean [hunk 1]
```diff
@@ -385,14 +385,14 @@ theorem map_eq_neg_iff_det_neg (x : Orientation R M ι) (f : M ≃ₗ[R] M)
 
 /-- If the index type has cardinality equal to the finite dimension, a basis with the given
 orientation. -/
-def someBasis [Nonempty ι] [DecidableEq ι] (x : Orientation R M ι)
+def someBasis [Nonempty ι] [DecidableEq ι] [Module.Finite R M] (x : Orientation R M ι)
```
Comment (sgouezel) [inline Mathlib/LinearAlgebra/Orientation.lean:388]:
```suggestion
def someBasis [Nonempty ι] [DecidableEq ι] [FiniteDimensional R M] (x : Orientation R M ι)
```
Not sure why you changed the formulation here.

File: Mathlib/LinearAlgebra/Orientation.lean [hunk 2]
```diff
@@ -385,14 +385,14 @@ theorem map_eq_neg_iff_det_neg (x : Orientation R M ι) (f : M ≃ₗ[R] M)
...
     (h : Fintype.card ι = finrank R M) : Basis ι R M :=
   ((finBasis R M).reindex (Fintype.equivFinOfCardEq h).symm).adjustToOrientation x
 
 /-- `some_basis` gives a basis with the required orientation. -/
 @[simp]
-theorem someBasis_orientation [Nonempty ι] [DecidableEq ι] (x : Orientation R M ι)
-    (h : Fintype.card ι = finrank R M) : (x.someBasis h).orientation = x :=
+theorem someBasis_orientation [Nonempty ι] [DecidableEq ι] [Module.Finite R M]
```
Comment (sgouezel) [inline Mathlib/LinearAlgebra/Orientation.lean:394]:
```suggestion
theorem someBasis_orientation [Nonempty ι] [DecidableEq ι] [FiniteDimensional R M]
```

File: Mathlib/MeasureTheory/Function/AEEqFun.lean [hunk 3]
```diff
@@ -260,7 +268,7 @@ theorem coeFn_comp (g : β → γ) (hg : Continuous g) (f : α →ₘ[μ] β) :
   rw [comp_eq_mk]
   apply coeFn_mk
 
-theorem comp_compQuasiMeasurePreserving [MeasurableSpace β] {ν} (g : γ → δ) (hg : Continuous g)
+theorem comp_compQuasiMeasurePreserving {β} [MeasurableSpace β] {ν} (g : γ → δ) (hg : Continuous g)
```
Comment (sgouezel) [inline Mathlib/MeasureTheory/Function/AEEqFun.lean:271]:
```suggestion
theorem comp_compQuasiMeasurePreserving {β : Type*} [MeasurableSpace β] {ν} (g : γ → δ) (hg : Continuous g)
```

File: Mathlib/MeasureTheory/Function/AEEqFun.lean [hunk 4]
```diff
@@ -401,13 +409,13 @@ theorem toGerm_injective : Injective (toGerm : (α →ₘ[μ] β) → Germ (ae 
   ext <| Germ.coe_eq.1 <| by rwa [← toGerm_eq, ← toGerm_eq]
 
 @[simp]
-theorem compQuasiMeasurePreserving_toGerm [MeasurableSpace β] {f : α → β} {ν}
+theorem compQuasiMeasurePreserving_toGerm {β} [MeasurableSpace β] {f : α → β} {ν}
```
Comment (sgouezel) [inline Mathlib/MeasureTheory/Function/AEEqFun.lean:412]:
```suggestion
theorem compQuasiMeasurePreserving_toGerm {β : Type*} [MeasurableSpace β] {f : α → β} {ν}
```

File: Mathlib/MeasureTheory/Function/AEEqFun.lean [hunk 5]
```diff
@@ -401,13 +409,13 @@ theorem toGerm_injective : Injective (toGerm : (α →ₘ[μ] β) → Germ (ae 
...
     (g : β →ₘ[ν] γ) (hf : Measure.QuasiMeasurePreserving f μ ν) :
     (g.compQuasiMeasurePreserving f hf).toGerm = g.toGerm.compTendsto f hf.tendsto_ae := by
   rcases g; rfl
 
 @[simp]
-theorem compMeasurePreserving_toGerm [MeasurableSpace β] {f : α → β} {ν}
+theorem compMeasurePreserving_toGerm {β} [MeasurableSpace β] {f : α → β} {ν}
```
Comment (sgouezel) [inline Mathlib/MeasureTheory/Function/AEEqFun.lean:418]:
```suggestion
theorem compMeasurePreserving_toGerm {β : Type*} [MeasurableSpace β] {f : α → β} {ν}
```

File: Mathlib/MeasureTheory/Measure/MeasureSpaceDef.lean [hunk 6]
```diff
@@ -265,7 +265,7 @@ disjoint unions, then the predicate holds for almost every `x : β` and all meas
 
 This is an AE version of `MeasurableSpace.induction_on_inter` where the condition is dependent
 on a measurable space `β`. -/
-theorem _root_.MeasurableSpace.ae_induction_on_inter {β} [MeasurableSpace β] {μ : Measure β}
+theorem _root_.MeasurableSpace.ae_induction_on_inter {α β} [MeasurableSpace β] {μ : Measure β}
```
Comment (sgouezel) [inline Mathlib/MeasureTheory/Measure/MeasureSpaceDef.lean:268]:
```suggestion
theorem _root_.MeasurableSpace.ae_induction_on_inter {α β : Type*} [MeasurableSpace β] {μ : Measure β}
```

File: Mathlib/RingTheory/Artinian.lean [hunk 7]
```diff
@@ -531,6 +531,7 @@ instance [IsReduced R] : DecompositionMonoid (Polynomial R) :=
 theorem isSemisimpleRing_of_isReduced [IsReduced R] : IsSemisimpleRing R :=
   (equivPi R).symm.isSemisimpleRing
 
-proof_wanted IsSemisimpleRing.isArtinianRing [IsSemisimpleRing R] : IsArtinianRing R
+proof_wanted IsSemisimpleRing.isArtinianRing (R) [CommRing R] [IsSemisimpleRing R] :
```
Comment (sgouezel) [inline Mathlib/RingTheory/Artinian.lean:534]:
```suggestion
proof_wanted IsSemisimpleRing.isArtinianRing (R : Type*) [CommRing R] [IsSemisimpleRing R] :
```

File: Mathlib/RingTheory/TensorProduct/MvPolynomial.lean [hunk 8]
```diff
@@ -187,7 +190,7 @@ noncomputable def scalarRTensorAlgEquiv :
   rTensorAlgEquiv.trans (mapAlgEquiv σ (Algebra.TensorProduct.lid R N))
 
 variable (R)
-variable (A : Type*) [CommSemiring A] [Algebra R A]
+variable {σ : Type*} (A : Type*) [CommSemiring A] [Algebra R A]
```
Comment (sgouezel) [inline Mathlib/RingTheory/TensorProduct/MvPolynomial.lean:193]: There is already a sigma around. Declaring another one is not a good idea.

File: Mathlib/Topology/Compactness/CompactlyGeneratedSpace.lean [hunk 9]
```diff
@@ -210,7 +211,7 @@ end UCompactlyGeneratedSpace
 
 section CompactlyGeneratedSpace
 
-variable {X : Type u} {Y : Type v} [TopologicalSpace X] [TopologicalSpace Y]
+variable {X : Type u} {Y : Type v} [tX : TopologicalSpace X] [tY : TopologicalSpace Y]
```
Comment (sgouezel) [inline Mathlib/Topology/Compactness/CompactlyGeneratedSpace.lean:214]: why this change?

File: Mathlib/Topology/VectorBundle/Hom.lean [hunk 10]
```diff
@@ -154,6 +155,7 @@ instance continuousLinearMap.isLinear [∀ x, ContinuousAdd (E₂ x)] [∀ x, Co
           simp_rw [smul_comp, comp_smulₛₗ, RingHom.id_apply]
           rfl }
 
+
```
Comment (sgouezel) [inline Mathlib/Topology/VectorBundle/Hom.lean:158]:
```suggestion
```

File: Mathlib/GroupTheory/PushoutI.lean [hunk 11]
```diff
@@ -563,7 +563,8 @@ noncomputable def equiv : PushoutI φ ≃ NormalWord d :=
       simp only [prod_smul, prod_empty, mul_one]
     right_inv := fun w => prod_smul_empty w }
 
-theorem prod_injective : Function.Injective (prod : NormalWord d → PushoutI φ) := by
+theorem prod_injective {ι} {G : ι → Type*} [(i : ι) → Group (G i)] {φ : (i : ι) → H →* G i}
```
[the remark this answers is not in the document]
Comment (semorrison) [reply Mathlib/GroupTheory/PushoutI.lean:566]: Sort* doesn't work.

File: Mathlib/Topology/Compactness/CompactlyGeneratedSpace.lean (hunk 9 above)
Comment (semorrison) [reply Mathlib/Topology/Compactness/CompactlyGeneratedSpace.lean:214]: I needed this for `theorem compactlyGeneratedSpace_of_coinduced` below, but I can do it locally.

File: Mathlib/RingTheory/TensorProduct/MvPolynomial.lean (hunk 8 above)
Comment (semorrison) [reply Mathlib/RingTheory/TensorProduct/MvPolynomial.lean]: Fixed by adding a section above.

File: Mathlib/RingTheory/Generators.lean [hunk 12]
```diff
@@ -215,6 +215,7 @@ def baseChange {T} [CommRing T] [Algebra R T] (P : Generators R S) : Generators
 end Construction
 
 variable {R' S'} [CommRing R'] [CommRing S'] [Algebra R' S'] (P' : Generators R' S')
+variable [Algebra R R'] [Algebra S S']
```
Comment (sgouezel) [inline Mathlib/RingTheory/Generators.lean:218]: This one looks wrong (and the other one in the same file also): you should'nt need to add new assumptions without removing other ones.

**Review by sgouezel — COMMENTED** [review]
bors d+

File: Mathlib/RingTheory/Generators.lean (hunk 12 above)
Comment (semorrison) [reply Mathlib/RingTheory/Generators.lean]: Fixed.

**Comment by semorrison** [comment]
bors merge
